src/components/EjectButton/EjectButton.js

```diff
@@ -17,30 +18,31 @@ type Props = {
   onClick: () => void,
 };
 
+export const dialogOptions: Electron.MessageBoxOptions = {
+  type: 'warning',
+  buttons: ['Yes, light this candle', "Ahhh no don't do that"],
+  defaultId: 1,
+  cancelId: 1,
+  title: 'Are you sure?',
+  message:
+    'Ejecting is a permanent one-time task that unwraps the create-react-app environment.',
+  detail:
+    "It's recommended for users comfortable with Webpack, who need to make tweaks not possible without ejecting.",
+};
+
+export function dialogCallback(response: number) {
+  // The response will be the index of the chosen option, from the
+  // `buttons` array above.
+  const isConfirmed = response === 0;
+
+  if (isConfirmed) {
+    this.props.onClick();
+  }
+}
+
 class EjectButton extends PureComponent<Props> {
   handleClick = () => {
-    dialog.showMessageBox(
-      {
-        type: 'warning',
-        buttons: ['Yes, light this candle', "Ahhh no don't do that"],
-        defaultId: 1,
-        cancelId: 1,
-        title: 'Are you sure?',
-        message:
-          'Ejecting is a permanent one-time task that unwraps the create-react-app environment.',
-        detail:
-          "It's recommended for users comfortable with Webpack, who need to make tweaks not possible without ejecting.",
-      },
-      (response: number) => {
-        // The response will be the index of the chosen option, from the
-        // `buttons` array above.
-        const isConfirmed = response === 0;
-
-        if (isConfirmed) {
-          this.props.onClick();
-        }
-      }
-    );
+    dialog.showMessageBox(dialogOptions, dialogCallback.bind(this));
```

Comment:
Nit: Maybe I'm inlining this in the render method as this got really short with the exports.